src/emacs/lean-server.el

```diff
@@ -44,8 +47,9 @@
        (setf (lean-server-session-tasks sess) res)
        (lean-server-notify-tasks-changed sess old-tasks)))
     ("error"
-     (message "error: %s" (plist-get res :message))
-     ; TODO(gabriel): maybe even add the error as a message
+     (when lean-server-print-errors-p
+       (message "error: %s" (plist-get res :message)))
```

Comment:
Ok, now I see the problem as well.  We always call `(message)` with the error, even if there is an error callback installed.  I think we should only call `(message)` if there is no error callback.